ui-tests/src/test/java/io/syndesis/qe/steps/other/JiraSteps.java

```diff
@@ -0,0 +1,152 @@
+import com.atlassian.jira.rest.client.api.domain.Comment;
+import com.atlassian.jira.rest.client.api.domain.Issue;
+import com.atlassian.jira.rest.client.api.domain.SearchResult;
+import com.atlassian.jira.rest.client.api.domain.input.IssueInputBuilder;
+import com.atlassian.jira.rest.client.api.domain.input.TransitionInput;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+
+public class JiraSteps {
+
+    private static final long BUG_ISSUE_TYPE = 10006L;
+    private static final int DONE_TRANSITION_TYPE = 41;
+
+    private final JiraRestClient jiraRestClient;
+
+    private String sharedIssueKey;
+
+    @Autowired
+    public JiraSteps(JiraRestClient jiraRestClient) {
+        this.jiraRestClient = jiraRestClient;
+    }
+
+    @Given("create a new jira issue in project \"([^\"]*)\"")
+    public void createNewIssue(String project) {
+        IssueInputBuilder issueBuilder = new IssueInputBuilder("MTP", BUG_ISSUE_TYPE);
+        issueBuilder.setSummary("test issue");
+        issueBuilder.setDescription("this is the test issue");
+
+        BasicIssue result = jiraRestClient.getIssueClient().createIssue(issueBuilder.build()).claim();
+        sharedIssueKey = result.getKey();
+    }
+
+    @Then("check new comment exists in previously created jira issue with text \"([^\"]*)\"")
+    public void checkCommentExists(String commentText) {
+        TestUtils.waitFor(() -> commentExists(sharedIssueKey, commentText),
+                1, 60,
+                "Could not find comment in jira issue");
+    }
+
+    private boolean commentExists(String issueKey, String text) {
+        Issue issue = jiraRestClient.getIssueClient().getIssue(issueKey).claim();
+        boolean commentFound = false;
+        for (Comment comment : issue.getComments()) {
+            if (StringUtils.equalsIgnoreCase(comment.getBody().trim(), text.trim())) {
+                commentFound = true;
+            }
+        }
+
+        return commentFound;
+    }
+
+    @Then("delete previously created issue")
+    public void deleteIssue() {
+        jiraRestClient.getIssueClient().deleteIssue(sharedIssueKey, true).claim();
+    }
+
+    @Then("close previously created issue")
+    public void closePreviousIssue() {
+        closeIssue(sharedIssueKey);
+    }
+
+    private void closeIssue(String issueKey) {
+        TransitionInput transitionInput = new TransitionInput(DONE_TRANSITION_TYPE);
+        Issue issue = jiraRestClient.getIssueClient().getIssue(issueKey).claim();
+        jiraRestClient.getIssueClient().transition(issue, transitionInput).claim();
+    }
+
+    @When("fill in issuekey for previously created issue")
+    public void fillIssueKey() {
+        Map<String, String> map = new HashMap<>();
+        map.put("issuekey", sharedIssueKey);
+
+        Form.waitForInputs(20);
+        new Form(new SyndesisRootPage().getRootElement()).fillByTestId(map);
```

Comment:
maybe skip the `map` variable and use `new Form(new SyndesisRootPage().getRootElement()).fillByTestId(TestUtils.map("issuekey", sharedIssueKey));` ?